---
title: "rRNA methylation in Paramxyovirus infection"
author: "Marina Alexander"
date: "29/11/2019"
output: 
  html_document:
    code_folding: hide
---

```{r setup, include=FALSE}

knitr::opts_chunk$set(
  echo = TRUE,
  warning = FALSE,
  message = FALSE,
  fig.align = 'center'
)


# remind R where to look for libraries
.libPaths(c("C:/Users/ale097/Data School/Packages"))
# load libraries
library(tidyverse)
library(knitr)
library(wesanderson)
library(kableExtra)
install.packages("unglue")
library(unglue)
# read in the files 
ribometh <- read_csv("data/Erales_2017_ribomethseq.csv")
snoRNAs <- read_csv("data/Jorjani_2016_snoRNA_atlas.csv")
hostgenes <- read_csv("data/Deffrasnes_2016_hostgenes_wildtypeHeV.csv")

snoRNAs %>%
  arrange(Name)

```

## snoRNAs involved in rRNA methylation by FBL in Hela cells

Erales et al. used Ribo-meth-seq to identify sites in the 28S rRNA subunit that were reduced in methylation by siFBL. Plotted here are 6 sites with a greater than 40% reduction from siNEG. I then downloaded data from the Jorjani snoRNA ATLAS to identify the snoRNAs that guide methylation to these sites. These are colored by their conservation with the order amniotes > tetropodes > vertebrates > deuterotomes most recent to most distantly evolved. Also from the Atlas I retrived their host gene. snoRNAs are "hosted" within introns of genes that are co-regulated in their transcription and usually act in the same pathway. 


```{r snoRNAs and their host genes that work with FBL}
# tidy data  
ribometh_tidy <- ribometh %>%
  mutate(mean_diff_percent = sub("%","", mean_diff )) %>% 
  mutate(mean_diff_percent = sub("-", "", mean_diff_percent)) %>% 
  mutate(mean_diff_percent = as.numeric(mean_diff_percent)) %>% 
  filter(SNORD != "NA")


selected_snoRNAs <- snoRNAs %>%
  select(Name, Expression, Conservation, `Host gene`) %>% 
  rename(SNORD = Name)

FBL_snoRNA <- left_join(ribometh_tidy, selected_snoRNAs, by = "SNORD")

write_csv(FBL_snoRNA,"results/FBL_SNORD_Hostgenes.csv") 

ggplot(FBL_snoRNA, aes(x= site, y= mean_diff_percent)) +
  geom_point()+
  theme(axis.text.x = element_text(angle = 90)) +
  geom_text(aes(label = SNORD, color = Conservation), hjust = 0.5, vjust = -0.5, size = 3) +
  geom_text(aes(label = `Host gene`), hjust = 0.5, vjust = 1.5, size = 2.5)+
  labs( y = "% change in methylation by siFBL",
        x = "rRNA site")


```

## RPL13 and RPL7A were hits in the screen
Deffrasnes et al identified 14 genes involved in protein translation that when knocked down reduced TCID50 of Hendra virus with a p-value <0.01. Listed here. 
```{r siRNA screen targets involved in translation}

hostgenes_tidy <- hostgenes %>% 
  filter(signif == "Yes") %>% 
  filter(Translation == "yes") %>% 
  arrange(TCID50)

knitr::kable(hostgenes_tidy, caption = 'Table 1. Translation genes required for Hendra virus infection')


```

```{r Anderson dataset}

Anderson_panviral <- read_csv("data/Anderson_ABCE1_panviral_hostgenes.csv")


```

```{r Complete modification map of human 80s ribosome}

RNAmodMap80S <- read_csv("data/Isobe_NAR_2018_80S_modifications_frequency.csv") %>% 
  rename(subunit = rRNA) %>% 
  rename(site = "Position assigned in this study")

FBL_snoRNA_modratio <- FBL_snoRNA %>% 
  unglue_unnest(site, "{mod=\\D+}{site}") %>%
  mutate(site = as.numeric(site)) %>% 
  left_join(RNAmodMap80S, by = "site") %>%
  rename(subunit = subunit.x ) %>%
  rename("host_gene" = "Host gene") %>% 
  rename("stoichiometry" = "Modification ratio") %>% 
  select(mod, site, mean_diff_percent, sd, stoichiometry, SNORD, host_gene)

write_csv(FBL_snoRNA_modratio,"results/FBL_SNORD_Hostgenes_modratio.csv") 
  
  

```

